src/dataflow/src/render/delta_join.rs

```diff
@@ -73,16 +75,54 @@ where
                             offset += arities[input];
                         }
 
+                        // filter arrangements using semijoins if possible
+                        let mut equivalences = equivalences.clone();
+                        for equivalence in equivalences.iter_mut() {
+                            equivalence.sort();
+                            equivalence.dedup();
+                        }
+
+                        // if possible, shorten input arrangements by joining them with a
+                        // constant row. Create a map to look up the shortened
+                        // arrangements to use
```

Comment:
I'm confused as to what is about to happen here. My understanding of delta joins is that they are invoked when we have all existing arrangements and do not need to create new ones. This description sounds like it is proposing introducing some new joins for the purposes of filtering, which looks like it will create a new arrangement in its implementation.